front/app/containers/SpamReport/index.tsx

```diff
@@ -56,21 +57,21 @@ const SpamReportForm = ({ resourceType, resourceId }: Props) => {
       return;
     }
 
-    setLoading(true);
-
-    sendSpamReport(resourceType, resourceId, diff)
-      .then(() => {
-        setLoading(false);
-        setSaved(true);
-        setErrors(null);
-        setDiff(null);
-      })
-      .catch((e) => {
-        setLoading(false);
-        if (isCLErrorJSON(e)) {
-          setErrors(e.json.errors);
-        }
-      });
+    addSpamReport(
+      { targetId, targetType, spam_report: diff },
+      {
+        onSuccess: () => {
+          setSaved(true);
```

Comment:
You could use the `isSuccess` helper from the query instead of this custom component state.